mne/time_frequency/tfr.py

```diff
@@ -181,10 +181,17 @@ def _centered(arr, newsize):
     return arr[tuple(myslice)]
 
 
-def _cwt_fft(X, Ws, mode="same", decim=1):
-    """Compute cwt with fft based convolutions
+def _cwt(X, Ws, mode="same", decim=1, use_fft=True):
+    """Compute cwt with fft based convolutions or temporal convolutions.
     Return a generator over signals.
     """
+    if mode not in ['same', 'valid', 'full']:
+        raise ValueError("`mode` must be 'same', 'valid' or 'full', "
+                         "got %s instead." % mode)
+    if mode == 'full' and (not use_fft):
+        # XXX JRK: full wavelet decomposition needs to be implemented
+        raise NotImplementedError('`full` decomposition with convolution is ' +
+                                  'currently not supported.')
```

Comment:
@agramfort Is it sufficient? L312 `'full' is currently not implemented with `use_fft=False`.`

> also please use ValueError

Ok